cmd/issue_edit.go

```diff
@@ -54,6 +55,34 @@ var issueEditCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
+		deleteNote, err := cmd.Flags().GetBool("deletenote")
+		if err != nil {
+			log.Fatal(err)
+		}
+		if deleteNote {
+			discussions, err := lab.IssueListDiscussions(rn, int(issueNum))
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			discussion := ""
+			for _, d := range discussions {
+				for _, n := range d.Notes {
+					if n.ID == commentNum {
+						discussion = d.ID
+						break
```

Comment:
This break will break the inner loop only, but the outer will keep running and, consequently, run the inner for the new discussions. You can either use a loop label or check if `discussion != ""` in the end of the outer loop.

Label example:
```go
myLabel:
    for _, d := range discussions {
        for _, n := range d.Notes {
            if n.ID == cummentNum {
                discussion = d.ID
                break myLabel
            }
        }
    }
```

The same happens in the mr_edit.go code.

Reply:
Fixed (and new version pushed).